kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/config/TaskDistributor.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.aws.s3.config
+
+import io.lenses.streamreactor.connect.aws.s3.config.S3ConfigSettings.TASK_INDEX
+
+import java.util
+import scala.jdk.CollectionConverters.SeqHasAsJava
+
+object TaskDistributor {
+  def distributeTasks(props: util.Map[String, String], maxTasks: Int): util.List[util.Map[String, String]] =
+    List.tabulate(maxTasks) { i =>
+      val map: util.Map[String, String] = new util.HashMap[String, String](props)
+      map.put(TASK_INDEX, s"$i:$maxTasks")
```

Comment:
This has to be a Java HashMap as it is used to pass the TASK_INDEX and other properties from one connect worker to the other and is serialised by the underlying Connect framework.  It would likely not handle the Scala map type.